network/linux_s/linux_s.c

```diff
@@ -0,0 +1,300 @@
+#include <netinet/in.h>
+#include <netdb.h>
+#include <string.h>
+#include <fcntl.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+static int32_t linux_socket_open(struct linux_desc *desc, uint32_t *sock_id,
+				 enum socket_protocol prot, uint32_t buff_size);
+static int32_t linux_socket_close(struct linux_desc *desc, uint32_t sock_id);
+static int32_t linux_socket_connect(struct linux_desc *desc, uint32_t sock_id,
+				    struct socket_address *addr);
+static int32_t linux_socket_disconnect(struct linux_desc *desc,
+				       uint32_t sock_id);
+static int32_t linux_socket_send(struct linux_desc *desc, uint32_t sock_id,
+				 const void *data, uint32_t size);
+static int32_t linux_socket_recv(struct linux_desc *desc, uint32_t sock_id,
+				 void *data, uint32_t size);
+static int32_t linux_socket_sendto(struct linux_desc *desc, uint32_t sock_id,
+				   const void *data, uint32_t size,
+				   const struct socket_address* to);
+static int32_t linux_socket_recvfrom(struct linux_desc *desc, uint32_t sock_id,
+				     void *data, uint32_t size,
+				     struct socket_address *from);
+static int32_t linux_socket_bind(struct linux_desc *desc, uint32_t sock_id,
+				 uint16_t port);
+static int32_t linux_socket_listen(struct linux_desc *desc, uint32_t sock_id,
+				   uint32_t back_log);
+static int32_t linux_socket_accept(struct linux_desc *desc, uint32_t sock_id,
+				   uint32_t *client_socket_id);
+
+
+
+struct network_interface linux_net = {
+	.socket_open = (int32_t (*)(void *, uint32_t *, enum socket_protocol,
+				    uint32_t)) linux_socket_open,
+	.socket_close = (int32_t (*)(void *, uint32_t)) linux_socket_close,
+	.socket_connect = (int32_t (*)(void *, uint32_t,struct socket_address *))linux_socket_connect,
+	.socket_disconnect = (int32_t (*)(void *, uint32_t))linux_socket_disconnect,
+	.socket_send = (int32_t (*)(void *, uint32_t, const void *, uint32_t))linux_socket_send,
+	.socket_recv = (int32_t (*)(void *, uint32_t, void *, uint32_t))linux_socket_recv,
+	.socket_sendto = (int32_t (*)(void *, uint32_t, const void *, uint32_t, const struct socket_address* to))linux_socket_sendto,
+	.socket_recvfrom = (int32_t (*)(void *, uint32_t, void *, uint32_t, struct socket_address* from))linux_socket_recvfrom,
+	.socket_bind = (int32_t (*)(void *, uint32_t, uint16_t))linux_socket_bind,
+	.socket_listen = (int32_t (*)(void *, uint32_t, uint32_t))linux_socket_listen,
+	.socket_accept= (int32_t (*)(void *, uint32_t, uint32_t*))linux_socket_accept
+};
+
+/**
+ * @brief Get network interface reference
+ * @param desc - Socket descriptor
+ * @param net - Address where to store the reference to the network interface
+ * @return
+ *  - \ref SUCCESS : On success
+ */
+int32_t linux_get_network_interface(struct linux_desc *desc,
```

Comment:
You can just use extern for linux_net  in linux_s.h like this iio example.
https://github.com/analogdevicesinc/no-OS/tree/master/iio/iio_ad9083

I was using wifi_get_network_interface because I needed to intialize the .net value.